### DISCUSSION TITLE
2024-03-01 - What is DNS - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by Dhruv-0987** [SSWConsulting/SSW.Rules.Content#8021] [Chewing the Fat]
Hey SSW'ers,

Let's talk about DNS.

https://www.ssw.com.au/rules/dns-what-and-how-it-works/

**Comment by MarcusIrmscherSSW** [comment]
As a non dev it was interesting to get a better understanding of DNS, its something that you encounter on a constant basis but never give it much thought.

**Comment by TobyChurches** [comment]
Great explanation and a clear rule. Although I've never needed to know this stuff (apart from a UNI exam) its good to have reference material for when the time arises.

**Comment by AlexRothwell** [comment]
Having a quick reference for what each type of DNS record is in the Rule is super handy. While I've had a vague understanding of how DNS fits together, its these kinds of details which have never really stuck for me and having it communicated so clearly is great

**Comment by joshbermanssw** [comment]
Video well accompanied the rule! + the style of CTF with small quiz like questions boosts engagement!

**Comment by calumjs** [comment]
Good to know.

**Comment by ricksu978** [comment]
Nice video with diagrams explaining how DNS work.

**Comment by alexleessw** [comment]
Good explanation!

**Comment by isaaclombardssw** [comment]
Video has great info on how DNS works, and the rule contains even more details making it a nice reference.

**Comment by jernejk** [comment]
Great refresher what we learned in high school. 😊
Although, I would probably have less issues configuring CloudFlare, if they would have taught as this well. 😂

**Comment by wicksipedia** [comment]
DNS, a haiku

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/600044/ff01431b-3f91-4497-a213-335748092c5e)

**Comment by YalalCakirSSW** [comment]
Good and clear explanation of DNS. I felt like I was in uni, studying for my finals again 😄

**Comment by zacharykeeping** [comment]
Good rule - very handy and informative.

**Comment by matt-goldman** [comment]
> It's always DNS

 _- Ancient Chinese proverb_

**Reply by chrisschultzssw** [reply]
beat me to it!

**Comment by Calinator444** [comment]
I learnt about DNS servers at uni and understood the role that DNS servers play in mapping domain names to IP addresses but this was a great refresher. I was a little unclear about the role root and top level domain name servers play in resolving domain name mappings beforehand.

Also fun fact, if you open up powershell and type in "nslookup www.google.com" you'll get the domain name mapping for google that's been stored by your operating system.

**Comment by awaltos** [comment]
Great rule + video. Sahn has a great way of explaining things in his videos. Perfect blend of animations and text making it easy to digest.

**Comment by lukecookssw** [comment]
`ipconfig /flushdns` has solved many problems in my life.

**Reply by 2fernandez** [reply]
ifconfig / -flushcache on a mac

**Comment by cvionnet** [comment]
The explanations in the rule are clear and a good addition to the video. Good job Dhruv 👍

**Comment by 2fernandez** [comment]
Awesome Rule and CTF @Dhruv-0987. I think it is well presented and the rule was well written.

A key takeaway for me was SRVs. I've not set one up before, and neither did I have the knowledge it existed. 

I was particularly interested in what the functionality behind an SRV is.
(More on SRV's here: https://www.cloudflare.com/en-au/learning/dns/dns-records/dns-srv-record/). 
I doubt I would use this in a DNS record, but it is still good to know.

Thanks

**Comment by Ben0189** [comment]
Good rules, i have learned in details of how DNS translate human readable domain name to ip address

**Comment by Harry-Ross** [comment]
Great video that succinctly explained the fundamentals of how a DNS resolver finds records. The rule was also a super convenient resource with all of the important info you would need, great for future reference. Both were great refreshers - in uni we had to build our own DNS resolver in Python, and that really drilled most of these concepts into my head.

**Comment by steven0x51** [comment]
Great explanation of DNS. Good video

**Comment by ozair-ashfaque** [comment]
Great explanation on DNS

**Comment by bettybondoc** [comment]
Great refresher on how DNS works. Well-explained video and rule. :)

**Comment by amankumarrr** [comment]
I enjoyed the animation and his explanation of such a complex topic; it made it really easy to understand.

**Comment by chrisschultzssw** [comment]
It's always DNS,
Guiding data's endless quest,
Silent, it impress. 
🤖

**Comment by AshAnil** [comment]
The video about DNS is really good. ⭐

**Comment by bahjatalaadelssw** [comment]
It's impressive to see how robust the global framework is for an infinitely scalable system. 🤯

**Reply by tomek-i** [reply]
the power of decentralisation :-D

**Comment by leahy268** [comment]
Been doing DNS since I've been a SysAdmin.  Pretty simple protocol that works really well.  Be aware that DNS changes can take up to 72 hours to fluctuate properly.

**Comment by tomek-i** [comment]
It's a good refresher and handy to know some DNS tools like mxtoolbox to check if records have been updated.

**Comment by christoment** [comment]
Great to learn the actual terminology for entities in the DNS resolution process 😄

**Comment by jeoffreyfischer** [comment]
That's the first time I've heard of DNS - I will dig a bit deeper to understand the concept better

**Comment by jimmidier** [comment]
Very well organized and explained rule, nice work

**Comment by jerryluossw** [comment]
Great rule, It makes me more clear about the process to translate this human-readable domain name into a machine-readable IP

**Comment by sylhuang** [comment]
Well explained. Learned something new from the video and the rule!

**Comment by AnkyChe** [comment]
Great rule! As a non-dev, it's interesting to learn.

**Comment by babakamyljanovssw** [comment]
Good explanation of DNS! Reminded me Computer Networking class at the uni 😀

**Comment by SofieHong** [comment]
Great video 🌟 Enjoying the explanatory animations

**Comment by KristenHu** [comment]
Great explanation, it brings back a lot of memories from my uni's lessons.

**Comment by tino-liu** [comment]
Great explanation! Need to learn a bit more about DNS though

**Comment by PothieuG** [comment]
ByteByteGo really do know how to explain things ! I really appreciate all the videos of his channel.

**Comment by Aibono1225** [comment]
Very clear explanation, learned many new points from the video and rules.

**Comment by AntPolkanov** [comment]
That's a great refresher! Can we please do a series of CTFs to go in depth?

**Comment by JackDevAU** [comment]
Great video - let a bunch about DNS 💪

**Comment by synnoveisabel** [comment]
Good video with nice diagrams, and very informative rule 😊

**Comment by MattParkerDev** [comment]
Nice to get a deeper understanding of how urls are resolved via DNS :)

**Comment by danielmackay** [comment]
This CTF filled in a few gaps I had around DNS.  For people who haven't had much exposure for this this before, I imagine it's might be overwhelming to realizes there is a complex process under the hood for something we all use everyday that "just works".

I think there were some good tips towards the end of the video, outlining some of the ways we can prep when needing to make an DNS update (e.g. lowering the TTL well in advance of making the update)

**Comment by jimmy1992abc** [comment]
8.8.8.8 is my only knowledge about DNS before, but getting more information thanks to the CTF today.

**Comment by tombui99** [comment]
Good rule and video - though as a dev, I have learnt about DNS long before but it's nice to get reminded

**Comment by Freego1783** [comment]
I'm sure I learnt about this in engineering school but it's nice having this kind of refresher from time to time! I'm rarely using this knowledge so I tend to forget 😁

**Comment by jakebayliss** [comment]
Interesting refresher. Great video and rule!

**Comment by stephanfakossw** [comment]
Nice rule, great explanation...

**Comment by vladislav-kir** [comment]
The rule is great, really easy to understand!

**Comment by drwharris** [comment]
Now lets talk about Secure DNS :)

**Comment by AttackOnMorty** [comment]
DNS  - Phone book of the Internet

**Comment by duanxinhuan** [comment]
Thanks!!!
I have learned something from this CTF!

**Comment by Marxoz** [comment]
Really good rule and video! Learned a few new things.

**Comment by SamBWagner** [comment]
Great rule, if a little dense.

**Comment by jackreimers** [comment]
Very informative video and rule

**Comment by PennyWalker** [comment]
It was interesting to see how everything connects! I never realised there was so much going on behind the scenes!
